frontend/src/shared/components/SignForm.tsx

```diff
@@ -0,0 +1,46 @@
+import { useCallback, FormEvent } from "react";
+import { Link } from "react-router-dom";
+
+type SignFormProps = {
+    onSubmit: (username: string, password: string) => void;
+    title: string;
+    submitText: string;
+    linkText: string;
+    linkTo: string;
+}
+
+
+export const SignForm = (props: SignFormProps) => {
+   const { onSubmit } = props; 
```

Comment:
[nits]
インデントなるべく揃えておきたい！

```suggestion
  const { onSubmit } = props;
```